contrib/babelfishpg_common/src/collation.c

```diff
@@ -1517,4 +1517,27 @@ babelfish_update_server_collation_name(PG_FUNCTION_ARGS)
 	server_collation_name = pstrdup(babelfish_restored_server_collation_name);
 	MemoryContextSwitchTo(oldContext);
 	PG_RETURN_VOID();
+}
+
+
+/*
+ * print_const_collation - determines whether collation needs to be dumped or not
+ * for collatable data types such as varchar, char, nvarchar, nchar, etc.
+ */
+bool
+print_const_collation(Const *constval) {
+    Oid typid = constval->consttype;
+    Oid constcollid = constval->constcollid;
+    Oid typcollation = get_typcollation(typid);
+
```

Comment:
why about other data types with const collation?

Reply:
This change is specifically useful for DEFAULT clause and check constraints. It will always have the const with base type. so checking for these 4 would be enough. 

We have the test case mentioned in BABEL-3801 and it works fine with this change